scripts/analyze_training_metrics.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import argparse
+import json
+import logging
+import sys
+from collections import defaultdict
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import Dict
+
+import matplotlib.dates as mdates
+import matplotlib.pyplot as plt
+
+LOGGER = logging.getLogger(__name__)
+
+logging.basicConfig(level=logging.INFO)
+
+
+def plot_graph(
+    model_name: str,
+    metric_name: str,
+    values_dict: Dict[datetime, float],
+    output_directory: Path,
+) -> bool:
+    sorted_metrics = sorted(values_dict.items())
+    x, y = zip(*sorted_metrics)
+
+    # Compute the threshold
+    if len(y) >= 2:
+        before_last_value = y[-2]
+    else:
+        before_last_value = y[-1]
+    ninety_five_percent = before_last_value * 0.95
```

Comment:
Nit: define the 0.95 in a constant at the top, and rename `ninety_five_percent` to use a generic name